### ISSUE TITLE
Closing Activity may still open browser (PayPal)

### ISSUE THREAD

**Issue by brudaswen** [braintree/braintree_android#295]
### General information

* SDK/Library version: 3.9.0
* Environment: -
* Android Version and Device: Not version specific
* Braintree dependencies: 
  - com.braintreepayments.api:braintree:3.9.0

### Issue description
- Create new `BraintreeFragment` and request PayPal billing agreement.
- Exit Activity while network request is still ongoing.
  - For me this means waiting roughly two seconds.
- **Actual:** The browser opens shortly after exiting.
- **Expected:** Browser should not open since PayPal request can not be handled successfully.

```
val brainTreeFragment = BraintreeFragment.newInstance(this, token)
PayPal.requestBillingAgreement(brainTreeFragment , PayPalRequest())
```

*PR is coming in the next minutes...*

**Comment by sshropshire** [comment]
Hey @brudaswen thanks for using the Braintree SDK for Android.

For PayPal, we're using an Android Intent to open the browser. Can you provide more information on this use-case in the context of your app?

**Comment by brudaswen** [comment]
Hi @sshropshire, thanks for your reply.

Our use-case is pretty normal and similar to the implementation in [braintreepayments/demo/PayPalActivity.java](https://github.com/braintree/braintree_android/blob/master/Demo/src/main/java/com/braintreepayments/demo/PayPalActivity.java).

I just tested with your Demo app and can reproduce the issue:
- Goto PayPal Activity.
- Click on "Billing Agreement".
- Click BACK (Android) -- this has to happen fast.
- The browser will now open shortly after you are on the start screen. (This should not happen!)
- However, the `BraintreeFragment` is already gone and finishing the PayPal flow will have no effect.

**Comment by brudaswen** [comment]
Here is a video that shows that the browser opens after leaving the PayPal Activity. Finishing the flow does not have a positive effect.

https://drive.google.com/file/d/1wAFzo_h6JqrIm6yWimBoZjh81-gsx6f3/view?usp=sharing

**Comment by sshropshire** [comment]
We appreciate your feedback @brudaswen we're looking into this issue!

**Comment by KanyindaK** [comment]
Hi there, 
Please I need assistance I can't run the current project. 
I'm getting the below error message :

Unable to load class 'javax.xml.bind.JAXBException'.
Possible causes for this unexpected error include:
Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)
Re-download dependencies and sync project (requires network)

The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.
Stop Gradle build processes (requires restart)

**Comment by scannillo** [comment]
Hi @KanyindaK - your issue seems irrelevant to the one opened in this issue tab. If you have found a bug in our SDK, file a new issue filing out the appropriate issue template. For help troubleshooting your specific integration, contact our Support team https://help.braintreepayments.com/.

**Comment by sshropshire** [comment]
Fixed as of `v3.11.0`.
